book/tutorials/nb-to-package/intro.md

```diff
@@ -0,0 +1,270 @@
+
+    ```python
+    def say_hello():
+        print("Hello, World!")
+    ```
+
+2. Then, you can import this module in another Python script or notebook:
+
+    ```python
+    from hello_world import say_hello
+
+    say_hello()
+    ```
+
+    This will print `Hello, World!` to the console.
+
+This is a simple example of a module, but you can have more complex modules with many functions and classes.
+
+![Bed](https://images.unsplash.com/photo-1522771739844-6a9f6d5f14af?q=80&w=3871&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D)
+
+Going back to the bedroom analogy, a module is like a single piece of furniture in a room.
+For example, a bed is a piece of furniture that we spend most of our life on,
+this "bed" might be made up of various parts like a headboard, frame, and mattress.
+Each of these parts can be thought of as a function or class in a module.
+
+### Library
+
+A library is a collection of modules and packages that provide specific functionality or a set of related functionalities.
+In the Python ecosystem, a library is often used interchangeably with the term "package", and this can be confusing.
+For example, [Matplotlib](https://matplotlib.org/) is a plotting library that contains many modules to help you plot data
+or [Xarray](http://xarray.pydata.org/en/stable/) is a library that includes several modules and packages to help you work with labeled multi-dimensional arrays.
+You can "include" these libraries in your program to add features without reinventing the wheel.
+
+![blueprint](https://images.pexels.com/photos/4792480/pexels-photo-4792480.jpeg)
+
+To make it clear, let's go back to the bedroom analogy.
+A library is like a collection of rooms in a house, it is beyond just a single room.
+It is the house itself, which contains many rooms.
+Different house or even a commercial space might have different purposes and functionalities.
+
+### The Python Terms Hierarchy
+
+Here's a hierarchy of Python terms from smallest to largest to help you understand the relationship between modules, packages, and libraries:
+
+<img src="../../img/code-to-package.png" alt="functions to libraries" width="200"/>
+
+A python package may look like the following:
+
+```console
+mypackage/
+    __init__.py
+    module1.py
+    module2.py
+    subpackage/
+        __init__.py
+        submodule1.py
+        submodule2.py
+
+```
+
+Within the `mypackage` package and its subpackages, you might import functions from a dependency like `numpy`
+within a module file. Once this set of packages is bundled together and distributed,
+you'll end up essentially with a library that can be installed and used by others.
+
+This action of bundling your code together into a package or library is called "packaging" your code.
+
+```{note}
+For the rest of this tutorial, we'll use the term "package" to refer to a collection of code that can be installed and used by others,
+which essentially means a **library**.
+```
```

Comment:
@scottyhq This is where I called out that the rest of the tutorial I'm using "package" for library! It is confusing in the Python world.. which is unfortunate, but I'd also like to drive home that a "package" is simply a directory with an `__init__.py` file and "library" is a collection of both packages and modules.

Reply:
All good. You could consider moving this hierarchy and note up to the top to define the terms and then jump into the analogy. I'm a visual person so I like diagrams :) But everyone's different. Happy to have you merge this as-is!